Framework/CMakeLists.txt

```diff
@@ -6,3 +6,4 @@ if(ARROW_FOUND)
 add_subdirectory(ArrowTests)
 endif()
 add_subdirectory(DebugGUI)
+add_subdirectory(Foundation)
```

Comment:
would prefer to have this in the top level, think it's something for the whole project (yes I know, can still be included)

Reply:
My worries is that if it is put in something like `Common` then it becomes no-man's land. Having it into `Framework` makes ownership clear at least.

Anyways, as long as we all agree no boost / ROOT will enter it (unlike other efforts at top level) I have no objection to move it out of `Framework`.